superduperdb/backends/base/query.py

```diff
@@ -1,761 +1,268 @@
-            pre_like=pre_like,
-            post_like=post_like,
-            query_linker=query_linker,
-        )
-
-    def _get_query_component(
-        self,
-        name: str,
-        type: str,
-        args: t.Optional[t.Sequence] = None,
-        kwargs: t.Optional[t.Dict] = None,
-    ):
-        query_component_cls = self.table_or_collection.query_components.get(
-            name, QueryComponent
-        )
-        return query_component_cls(name, type=type, args=args, kwargs=kwargs)
-
-    @abstractmethod
-    def _get_query_linker(cls, table_or_collection, members) -> 'QueryLinker':
-        pass
-
-    def __getattr__(self, name):
-        _check_illegal_attribute(name)
-        assert self.post_like is None
-        if self.query_linker is not None:
-            query_linker = getattr(self.query_linker, name)
+    def unique_id(self):
+        return self.id
+
+    def _deep_flat_encode(self, cache, blobs, files, leaves_to_keep=()):
+        query, documents = self._dump_query()
+        cache[self.id] = {
+            '_path': 'superduperdb.backends.base.new_query.parse_query',
+            'documents': documents,
+            'query': query,
+        }
+        return self.id
+
+    @staticmethod
+    def _update_item(a, documents, queries):
+        if isinstance(a, Query):
+            a, sub_documents, sub_queries = a._to_str()
+            documents.update(sub_documents)
+            queries.update(sub_queries)
+            id_ = uuid.uuid4().hex[:5].upper()
+            queries[id_] = a
+            arg = f'query[{id_}]'
         else:
-            query_linker = self._get_query_linker(
-                self.table_or_collection,
-                members=[self._get_query_component(name, type=QueryType.ATTR)],
+            id_ = uuid.uuid4().hex[:5].upper()
+            if isinstance(a, dict):
+                documents[id_] = a
+                arg = f'documents[{id_}]'
+            elif isinstance(a, list):
+                documents[id_] = {'_base': a}
+                arg = f'documents[{id_}]'
+            else:
+                try:
+                    arg = json.dumps(a)
+                except Exception:
+                    documents[id_] = {'_base': a}
+                    arg = id_
+        return arg
+
+    def _to_str(self):
+        documents = {}
+        queries = {}
+        out = self.identifier[:]
+        for part in self.parts:
+            if isinstance(part, str):
+                out += f'.{part}'
+                continue
+            args = []
+            for a in part[1]:
+                args.append(self._update_item(a, documents, queries))
+            args = ', '.join(args)
+            kwargs = {}
+            for k, v in kwargs.items():
+                kwargs[k] = self._update_item(v, documents, queries)
+            kwargs = ', '.join([f'{k}={v}' for k, v in kwargs.items()])
+            if part[1] and part[2]:
+                out += f'.{part[0]}({args}, {kwargs})'
```

Comment:
I reckon this is for following

`table.filter(table.x == 1).select(...)`

is there be a possibility of ternary expression ? if yes, will the above suffice?